### ISSUE TITLE
Problem: inconsistent naming of "account" vs "staked state"

### ISSUE THREAD

**Issue by tomtau** [crypto-com/chain#1184]
Currently, the source code (and perhaps some documentation?) uses them interchangeably.
It'll be better to make the naming consistent.

One reason for "staked state" was that "account" may be a bit confusing, as some people may associate it with "bank account" or "Eth account", but here it's quite different: 1) it has two "balances", 2) it has some extra metadata, 3) its operations are restricted to staking / network-related functionality, 4) there's no "account-to-account": https://crypto-com.github.io/getting-started/transaction-accounting-model.html

**Comment by yihuang** [comment]
Lots of "account" just in the docs link you mentioned = =", I think the account is more familiar to most people.

**Comment by lezzokafka** [comment]
> Lots of "account" just in the docs link you mentioned = =", I think the account is more familiar to most people.

It happens all the time..... 

" I've just deposit the funds into your staking address" 
   "..? I don't see it in my balance. " 
"It is a staking address, so please check the state of your address" 
   "..? I have check my balance, It shows zero "

**Comment by calvinlauco** [comment]
How about we distinguish by "Staking Account" and "Wallet". "Account" is too general for most people. Though for @lezzokafka  case you probably need to point them which RPC/command to use.

**Comment by yihuang** [comment]
> How about we distinguish by "Staking Account" and "Wallet". "Account" is too general for most people. Though for @lezzokafka case you probably need to point them which RPC/command to use.

I agree, "Transfer account" and "Staking account", should be clear to the user.
Also, if an API named "account" without specifying transfer or staking, we should expect it to work on both.

For @lezzokafka's case, we can either let "balance" shows both transfer and staking balance, or rename it into "transfer_balance/transfer_account", and "staking_state" can be renamed into "staking_account", for consistency.

**Comment by tomtau** [comment]
"Transfer account" may again be confusing, as there's no account -- e.g. no-one talks about "bitcoin accounts". "wallet" may be a bit more familiar.
but again, then that's overloaded, as the "high-level" thing is called "wallet".

What about if the "high-level" grouping is called something else -- e.g. "vault". 

"vault"->"wallet"->"transfer balance" (or could be called "spendable balance", "wallet balance" or something like that)
"vault"->"staking account"->"staking balance"

(that's just logical grouping -- there could still be a single balance, that shows both)

"client-*" could then be renamed based on that -- e.g. "chain-vault-manager" -- https://github.com/crypto-com/chain/issues/812

**Comment by yihuang** [comment]
I think:

- "Wallet" is fine to cover both UTxO model and account model(I think we can still be called account-model even if not spendable), since it's already used by everyone for utxo model, accout model, also PoS staking stuff.
- So we can have like "wallet balance" for both UTxO and staked assets, "wallet balance --transfer" for UTxO balance, "wallet balance --staking" for staked part of the assets.
  Or "wallet balance" still displays UTxO balance by default, support "wallet balance --staking" to list all non-zero staked assets.
  This should be enough to solve the user's confusion.

For our own usage in docs and code:
- I prefer stake to account, account indeed sounds vague.
- But I don't like "StakedState", because it's two words, and why the "ed" since "stake" is already a noun, I'm not an expert on English though.

Just my two cents.

**Comment by yihuang** [comment]
How about just using `staking`:
- We already use it in the client: `address list --staking`.
- `struct Staking { bonded, ...`
- `chain-core/src/state/staking.rs`
- `StakingAddress`, `StakingStorage`

Looks quit consistent to me.

**Comment by lezzokafka** [comment]
>We can have like "wallet balance" for both UTxO and staked assets, "wallet balance --transfer" for UTxO balance, "wallet balance --staking" for staked part of the assets.
 
I also prefer`wallet balance --transfer` / `wallet balance --staking`, but would have to go a bit deeper with the design.

Let  `{T_i}` and `{S_i}` be those Transfer and Staking addresses under the wallet `W` :

-  `wallet balance --transfer ` would give you the the sum of the unspend UTXOs of all `{T_i}`(total/pedning/avaiable)

-  `wallet balance --staking` would show all the staked assets in the collection of `S_i` ?

It works fine for the UTXOs case but thoses staking address would be, in a sense, independent from each other?

**Comment by yihuang** [comment]
> It works fine for the UTXOs case but thoses staking address would be, in a sense, independent from each other?

Since they belong to the same wallet, so in a sense, they are related.
Suppose some users ask "where are my coins gone", you can instruct him/her to execute "wallet balance --staking".
The implementation will be something like this:
```
for addr in `address list --staking` {
  let state = `staking state $addr`
  if state.bonded > 0 or state.unbonded > 0 {
    print state
  }
}
```
